util/ibex_config.py

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/env python3
+
+import argparse
+import collections
+import os
+import sys
+
+import hjson
+
+DEFAULT_CONFIG_FILE = 'ibex_configs.hjson'
+
+
+class ConfigException(Exception):
+    pass
+
+
+def verify_config(name, config_dict):
```

Comment:
I'll read it and have a think.

Reply:
The easier and more scalable option is using an existing schema validator. For JSON and YAML that would be jsonschema. But in general I wouldn't invest much into validating here. It's a trivial script and already quite "blown up" by infrastructure around its primary goal.

Reply:
So I've basically left the validation as is, I feel the structure is actually what @rswarbrick link is getting at in principle, the verify functions are intended to be internal, so any external user must use the 'parse' function (as the link would put it) which either gives you a valid configuration or results in an exception.

Briefly considering schema validation but haven't implemented for two reasons:
1. Will spend a while working out how to do it and I'm trying to avoid too much extra effort on this
2. From my brief reading it wasn't obvious how to enforce the all configurations must have the same parameter set rule via schemas. May be possible but see point 1.